### PR TITLE
Update/remove daily slack summary

### PR DESCRIPTION
**Pull request by rodelgc**
## Description

This PR removes this old Slack summary for the daily test results:

GRVAh5XLpw

Because we now have this new Slack summary in place:

Ct9FjJshIk

In order to remove it, this PR makes the following changes:
- Remove the steps in the `publish-test-reports-daily.yml` workflow that are related to sending the old Slack summary.
- Remove these two files because they're not needed anymore:
    - `.github/workflows/publish-test-reports-daily.yml`
    - `.github/workflows/scripts/templates/SLACK_DAILY.json`

This PR also disables these steps when the `publish-test-reports-daily.yml` workflow was triggered by a non-`trunk` branch, so that stats won't be affected:
- Use JSON key file
- Update e2e test suite stats
- Update API test suite stats

I already deleted the `SLACK_CHANNEL_ID_DAILY` GitHub secret since it's no longer needed. This resulted to a failing _"Post test report to Slack channel"_ step like this [one](https://github.com/woocommerce/woocommerce-test-reports/actions/runs/6294653080/job/17086788209#step:19:19) in `trunk`. But I believe this failure is fine, since we have to test this PR without the `SLACK_CHANNEL_ID_DAILY` secret to make sure the `publish-test-reports-daily.yml` workflow still works without it.

## Testing instructions
1. Go to [Smoke test daily](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-daily.yml).
1. Choose a run from the past 5 days to make sure their artifacts haven't expired yet, and click on it.
1. Make sure that the API and E2E artifacts are not expired yet. If they are, choose a different, more recent run.
1. Take note of the following:
    - Run ID
        7fY4G2cvfQ
    - The API and E2E artifact names.
        BI57JUwbdu
1. Go to [Publish Daily smoke test report](https://github.com/woocommerce/woocommerce-test-reports/actions/workflows/publish-test-reports-daily.yml)
1. Click on the `Run workflow` menu.
1. Select this branch ( `update/remove-daily-slack-summary` )
1. Enter the run ID in `Daily smoke test workflow run ID.`
1. Enter the API and E2E artifact names in the `API test report artifact.` and `E2E test report artifact.` inputs.
1. In the `Root prefix of the test report.` field, leave the default value `test`. DO NOT change the value to `public`.
1. Click on the green `Run workflow` button.
1. Wait for the workflow to finish.
1. Verify that the "Publish Daily smoke test report" run you triggered passes.
1. Verify that these three steps were skipped:
    - Use JSON key file
    - Update e2e test suite stats
    - Update API test suite stats
1. Verify that the old Slack summary wasn't sent in the `#woo-smoke-tests-daily` Slack channel.
1. Verify that you can access the API and E2E Allure reports in:
    - https://a8c-woo-test-reports.s3.amazonaws.com/test/daily/api/index.html
    - https://a8c-woo-test-reports.s3.amazonaws.com/test/daily/e2e/index.html

### CODE REVIEW

**Comment by alopezari** [comment]
Hi @rodelgc! Thanks for creating this PR and improving the Test Report Slack message, it looks way simpler now even with better coverage, that's amazing!

I performed the steps listed in the testing instructions and everything works as expected except for the step 13:
`Verify that the "Publish Daily smoke test report" run you triggered passes.`, which failed because the Channel Id was missing, as you mentioned in the testing instructions, so I assume the behaviour is correct and it's a small gap in the testing instructions (probably because these were written before the Channel Id variable was deleted).

In a nutshell, it failed because the `Post test report to Slack channel` job was using the removed Channel Id, but once the PR is merged, that step will no longer exist, so that won't be a problem.

Also, I've checked that the `SLACK_CHANNEL_ID_DAILY` is no longer used anywhere in the repository, so everything should be fine.

I'll approve this, thank you again!

I won't merge it just in case @lanej0 wants to take a look too 👍

**Review by alopezari — APPROVED** [review]

**Comment by rodelgc** [comment]
Thanks for taking a look @alopezari 🙇 

> I performed the steps listed in the testing instructions and everything works as expected except for the step 13:
Verify that the "Publish Daily smoke test report" run you triggered passes., which failed because the Channel Id was missing

I think I know why it failed. Maybe you triggered it with `trunk` selected as the branch? Would you be able to test again but this time selecting `update/remove-daily-slack-summary` as the branch?

**Comment by alopezari** [comment]
> I think I know why it failed. Maybe you triggered it with `trunk` selected as the branch? Would you be able to test again but this time selecting `update/remove-daily-slack-summary` as the branch?

Yes! That was it, I re-ran it again in this branch and [it passed](https://github.com/woocommerce/woocommerce-test-reports/actions/runs/6297668889) 👍

**Comment by lanej0** [comment]
No concerns from me.  The stats collection piece takes place outside of the Slack messaging, so there shouldn't be any impact. 👍🏼
